### PR TITLE
Fix resumed downloads not being updated in the UI

### PR DESCRIPTION
**Pull request by amlcurran**
Resuming an item would not change its state from paused to resume in UI. This was because we failed to notify a change from the batches URI. 

As another nice thing, we only attempt to resume items which aren't complete, which reduces flickering and reduces the amount of threads we create

### CODE REVIEW

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/158/
Test PASSed.

**Comment by amlcurran** [comment]
(I attempted to write a test but ContentValues tripped me over)

File: library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java [hunk 1]
```diff
@@ -446,17 +447,24 @@ public void pauseBatch(long id) {
     }
 
     public void resumeBatch(long id) {
-        ContentValues values = new ContentValues();
-        values.put(DownloadContract.Downloads.COLUMN_CONTROL, DownloadsControl.CONTROL_RUN);
-        values.put(DownloadContract.Downloads.COLUMN_STATUS, DownloadStatus.PENDING);
+        ContentValues values = getResumeContentValues();
         String where = COLUMN_BATCH_ID + "= ? AND " + DownloadContract.Downloads.COLUMN_STATUS + " != ?";
-        String[] selectionArgs = {String.valueOf(id), String.valueOf(DownloadStatus.PENDING)};
+        String[] selectionArgs = {String.valueOf(id), String.valueOf(DownloadStatus.SUCCESS)};
```
Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java]: https://github.com/novoda/download-manager/commit/6464ebd3dbf43d0517829fcf1ac16190d448e3ef

>4 days ago

#awkward

Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java]: nah but seriously, how come this has changed?

from the PR desc. it seems like the fix was adding the `notifyBatchesHaveChanged` method?

Comment (amlcurran) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java]: > we only attempt to resume items which aren't complete, which reduces flickering

Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java]: wouldn't we need to do both? `STATUS != SUCCESS AND STATUS != PENDING`?

Comment (amlcurran) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java]: Doesn't really matter if they're pending and you set them to pending again.

Also @zegnus commit message says to update non-complete items only but it updated non-pending items - was this intentional?

Comment (zegnus) [inline library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java]: Seems ok, my fix was to ensure that we only change the minimum number of values, added the STATUS != PENDING won't change the result but will avoid to change some unnecessary rows.

I'm ok with this

**Comment by zegnus** [comment]
:+1: LGTM

**Pull request merged**